### DISCUSSION TITLE
Timeouts when exporting/importing data or doing bulk updates

### DISCUSSION THREAD

**Discussion by empusas** [netbox-community/netbox-docker#1137] [Getting Help]
We have switched to the docker version of netbox as this is much easier to maintain as the native installation.
As we need to have SSL encryption we have a NGINX in from of netbox. I followed the recommendation to not mess around with the NGINX inside the netbox container.

But we are constantly having issues with timeouts when we export bigger amounts of data or try to import more then 20 lines of data. Same also when we bulk edit more than 100 items at once. So i tried to increase the timeout in our NGINX installation. But it did not improve the situation.

ngnix config:

```
{
    listen [::]:443 ssl ipv6only=off;
 
    # CHANGE THIS TO YOUR SERVER'S NAME
    server_name [*****](http://*****/);
 
    ssl_certificate /etc/ssl/certs/netbox.crt;
    ssl_certificate_key /etc/ssl/private/netbox.key;
 
    proxy_connect_timeout 600;
    proxy_send_timeout 600;
    proxy_read_timeout 600;
    send_timeout 600;
 
    client_max_body_size 25m;
 
    location /static/ {
        alias /opt/netbox/netbox/static/;
    }
 
    location / {
        proxy_pass http://127.0.0.1:8001/;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}
```

I have checked the error log of our NGNIX instance and see that there is clearly a timeout problem. Here i tried to export all devices, just bit less than 10.000 :

> 2024/01/18 20:04:43 [error] 2737062#0: *57524 upstream timed out (110: Connection timed out) while reading response header from upstream, client: ::ffff:15.211.60.90, server: ******, request: "GET /dcim/devices/?export HTTP/1.1", upstream: "http://127.0.0.1:8001/dcim/devices/?export", host: "******", referrer: [https://******/dcim/devices/](https://%2A%2A%2A%2A%2A%2A/dcim/devices/)

I also checked the logs form the netbox container log and see a 404 there as well:

> netbox-docker-netbox-1  | 172.20.0.1 - - [18/Jan/2024:19:18:58 +0000] "GET /dcim/devices/nginx-logo.png HTTP/1.0" 404 88420 "[https://*******/dcim/devices/?export](https://********/dcim/devices/?export)" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36 Edg/120.0.0.0"server 

I am a bit out of clues what I could tweak to solve those timeout issues. My next attempt would be to tinker with the config of the internal nginx, but maybe someone had the same problem and solved it?

**Comment by jonaagenilsen** [comment]
Same problem here. A lot more data though, tried to export changelog with 1,5M entries ;)

Browser gets a timeout but server still processing. My Nginx is identical.

![image](https://github.com/netbox-community/netbox-docker/assets/40994635/576aa10c-41dc-498c-973a-660870a279a7)
